.github/workflows/ci.yml

```diff
@@ -129,3 +144,87 @@ jobs:
           labels: ${{ steps.meta.outputs.labels }}
           cache-from: ${{ steps.registry_refs.outputs.tags }},mode=max
           cache-to: ${{ steps.registry_refs.outputs.tags }},mode=max
+
+  integration-tests:
+    needs: [determine-changes, lint, test, build-container]
+    runs-on: ubuntu-latest
+    if: ${{ needs.determine-changes.outputs.non-test-changes }}
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Extract Branch Name
+      run: |
+        if [ "$GITHUB_HEAD_REF" != "" ]; then
+          echo "BRANCH_NAME=$GITHUB_HEAD_REF" >> $GITHUB_ENV
+        else
+          echo "BRANCH_NAME=${GITHUB_REF#refs/heads/}" >> $GITHUB_ENV
+        fi
+
+    - name: Pull Public Docker Image from GHCR
+      run: |
+        docker pull ghcr.io/sage-bionetworks/genie:${{ env.BRANCH_NAME }}
+
+    - name: Start Docker Container
```

Comment:
Just FYI, I saw https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/running-jobs-in-a-container. It seems to be a way to spin up the container.

Reply:
Thanks I think I saw that. I'll take a look once I resolve the data discrepancies.

Reply:
Thanks for linking this! 

It looks like the github integrated container is much faster/just not using `docker exec` is the key. After some research, looks like `docker exec` causes extra latency and overhead and repeated calls builds up and just increase the slowness. Known issue: https://github.com/docker/for-linux/issues/466

Reply:
After attempting to set it up, I think there's a lot more to be done to auto-integrate in the nextflow run (like how to pull in the nextflow compute env programmatically since it changes via github actions) or debugging it (the error messages for the action before you even have a run going are not helpful because there are none). Will add my findings to the relevant JIRA ticket that has us integrate in a nextflow run.

Reply:
> Been trying out the github integrated container that you linked and it gives [similar run times](https://github.com/Sage-Bionetworks/Genie/actions/runs/13154694462) to the current layout here after a couple of runs (ignore the error message)
I'm not sure how to check the run time since the attached one is failed. I saw one [succeed job](https://github.com/Sage-Bionetworks/Genie/actions/runs/13151347465) takes 1.5hr which is a bit of concern.

Reply:
Yes, using Tower can at least avoid the CPU limitation in GH action. We might be able to use compute_env to specify desired env and [actions/upload-artifact](https://github.com/actions/upload-artifact) to save verbose log somewhere.

Reply:
Yeah so here are the previous integration test run times: https://github.com/Sage-Bionetworks/Genie/actions?query=workflow%3Abuild+branch%3Agen-1653-add-integration-tests+is%3Asuccess

If you click on the jobs, you'll see that for each step run, it is random which step takes a lot of time. In the failed run, you can see that just the processing of non-mutation data takes 30+ min.

Reply:
Yeah so since our nextflow compute_env always gets updated, we have to find a way to pull it programmatically  in GH actions (it's not very self-explanatory). Yes [actions/upload-artifact](https://github.com/actions/upload-artifact) starts working once the nextflow run launch is successful and kicks in.

Reply:
I think given the long runtimes, if it's a big concern. I can just close this PR and mark as `Won't Do`, and we can escalate this one: https://sagebionetworks.jira.com/browse/GEN-1658

It's also odd because the initial test runs of this were only 30 min for the whole pipeline.

Reply:
Since you've done quite a bit of work here already, I'm happy to merge this first and we can iterate on the solution. 

Sure these are long test times, but it does give us a starting point.